youtube_dl/options.py

```diff
@@ -298,6 +298,10 @@ def _scrub_eq(o):
         metavar='DATE', dest='dateafter', default=None,
         help='Download only videos uploaded on or after this date (i.e. inclusive)')
     selection.add_option(
+        '--date-ordered-playlist',
```

Comment:
Option should be called something like `--date-playlist-order` and should accept `asc`, `desc` or `none`. Code should process playlist according to it.